mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -2775,12 +2750,6 @@ int main(int argc, char **argv) {
     });
 
   } else {
```

Comment:
Remove this check, to allow clone-verification for generated-kernel cases too.  The old code didn't try to make the types work out, but now we make both kernel and validation function have the same types.